scripts/API.py

```diff
@@ -0,0 +1,438 @@
+	"""
+	# Getting current branch
+	currentBranch = getCurrentBranch(dir=dir)
+
+	# Check if previous command succeeded
+	if currentBranch is None:
+		return False
+
+	# Update branch
+	retCode = runInsistentJob("git fetch")[0]
+
+	# Check if command succeeded
+	if retCode != 0:
+		return False
+
+	# Get latest local commit
+	localCommit = runJob(f"git rev-parse {currentBranch}")[1]
+
+	# Get latest remote commit
+	retCode, remoteCommit = runInsistentJob(
+		f"git rev-parse origin/{currentBranch}")
+
+	# Check if command succeeded
+	if retCode != 0:
+		return False
+
+	# Return commit comparison
+	return localCommit == remoteCommit
+
+
+def runInsistentJob(cmd: str, cwd: str = './', showOutput: bool = False,
+                    showError: bool = False, showCommand: bool = False,
+                    nRetries: int = 5) -> Tuple[int, str]:
+	"""
+	### This function runs the given network command and retries it the number given of times until one of the succeeds or it fails for all the retries.
+
+	#### Params:
+	- cmd (str): Command to run.
+	- cwd (str): Optional. Path to run the command from. Default is current directory.
+	- showOutput (bool): Optional. If True, output is printed.
+	- showError (bool): Optional. If True, errors are printed.
+	- showCommand (bool): Optional. If True, command is printed in blue.
+	- nRetries (int): Optional. Maximum number of retries for the command.
+
+	#### Returns:
+	- (int): Return code.
+	- (str): Output of the command, regardless of if it is an error or regular output.
+	"""
+	# Running command up to nRetries times (improves resistance to small network errors)
+	for _ in range(nRetries):
+		output=[]
+		retCode = runJob(cmd, cwd=cwd, log=output, show_output=False,show_command=False)
+		# Break loop if success was achieved
+		if retCode == 0:
+			break
+	if len(output) > 0:
+		output = output[0]
+	# Enforce message showing deppending on value
+	if showCommand:
+		print(cmd)
+	if showOutput:
+		print('{}\n'.format(output))
+	if showError:
+		print(output)
+
+	# Returning output and return code
+	return retCode, output
+
+
+
+
+'''
+curl --header "Content-Type: application/json" -X POST --data '{"user": {"userId": "curl attempt"}, "version": {"os": "Ubuntu 22.04.3 LTS", "architecture": "skylake", "cuda": "11.4", "cmake": "null", "gcc": "10.5.0", "gpp": "10.5.0", "scons": "null"}, "xmipp": {"branch": "Unknow", "updated": false}, "returnCode": 0, "logTail": "null"}' --request POST xmipp.i2pc.es/api/attempts/
+
+
+curl --header "Content-Type: application/json" -X POST --data '{"user": {"userId": "f0ccfe1cac91db754d039cf3bb5e7f46327ef3c5442d245b4c5e3b28086003c5"}, "version": {"os": "Ubuntu 22.04.3 LTS", "architecture": "skylake", "cuda": "11.4", "cmake": "null", "gcc": "10.5.0", "gpp": "10.5.0", "mpi": null, "python": null, "sqlite": null, "java": null, "hdf5": null, "jpeg": null}, "xmipp": {"branch": "Unknow", "updated": false}, "returnCode": 0, "logTail": null}' --request POST xmipp.i2pc.es/api/attempts/
+
+'''
```

Comment:
Commented curl command, probably left here as a temporary note of what command should this code emulate as an example, but, if it already works, it can be (should be) removed

Reply:
I agree. I did not see this previously

Reply:
ups, thanks